packages/core-typings/src/ILivechatVisitor.ts

```diff
@@ -16,19 +16,35 @@ export interface ILivechatVisitorConnectionData {
 	clientAddress: string;
 }
 
+export enum VerificationStatusEnum {
+	InProgress = 'InProgress',
+	Off = 'Off',
+	UnVerified = 'UnVerified',
+	Verified = 'Verified',
+}
 export interface IVisitorEmail {
 	address: string;
+	verified?: VerificationStatusEnum;
```

Comment:
Let's revisit this. After giving this an another look, I see some potential issues if we store the status of verification on Visitor level. A visitor can start multiple chats at the same time and there can be a case where the visitor only completes the verification on one of the chat's. In this case, we should only show the room where the visitor has completed the verification as an verified room. Rest of the opened chats should still be considered unverified.

Let me know your thoughts on this or have further questions